parsel/selector.py

```diff
@@ -91,7 +91,10 @@ def re(self, regex):
         Call the ``.re()`` method for each element in this list and return
         their results flattened, as a list of unicode strings.
         """
-        return flatten([x.re(regex) for x in self])
+        results = list()
+        for x in self:
+            results += x.re(regex)
+        return results
```

Comment:
https://gist.github.com/Tethik/a798053feb0b2066bdebb00325e0cf8b
**NOW** is with the change you suggested. I cut out the benchmarks not running that part of the code. It does not seem to make much of a difference. 

I'd expect both ways of appending to the list to run at amortized O(1), but I don't know what python does under the hood for its list comprehensions.